### PR TITLE
chore: bump pyyaml to 5.3

### PR DESCRIPTION
**Pull request by joshuarli**
[PyYAML 5.1+ supports Python 3.6+](https://github.com/yaml/pyyaml/blob/master/CHANGES#L50). We only seem to need to load yaml for configuration related things, which seem to have decent test coverage, so this bump should be easy.

The only thing is that PyYAML 5.1 considers the valid yaml `{foo:bar}` to be actually valid, compared to < 5.1. I've updated tests to reflect this.

Also, it seems we use libyaml in prod - I ran tests locally against a pyyaml wheel with and without the built yaml SO, and observed no difference in results. But would it be worth adding libyaml to travis?

### CODE REVIEW

**Comment by joshuarli** [comment]
Sigh, tests are failing

```
__________________________ OptionsTypesTest.test_dict __________________________

Traceback (most recent call last):

  File "/home/travis/build/getsentry/sentry/tests/sentry/utils/test_types.py", line 76, in test_dict

    assert Dict("{foo:bar}") == {"foo:bar": None}

  File "/home/travis/build/getsentry/sentry/src/sentry/utils/types.py", line 39, in __call__

    raise InvalidTypeError(u"{!r} is not a valid {}".format(value, repr(self)))

InvalidTypeError: '{foo:bar}' is not a valid dictionary

 generated xml file: /home/travis/build/getsentry/sentry/.artifacts/python.junit.xml 
```

But this only applies to 3.13. Need to nuke travis cache.

File: src/sentry/utils/types.py [hunk 1]
```diff
@@ -116,7 +116,7 @@ class StringType(Type):
 
 
 class DictType(Type):
-    """Coerce a dict out of a json/yaml string"""
+    """Coerce a dict out of a yaml string"""
```
Comment (mattrobenolt) [inline src/sentry/utils/types.py]: I'd argue to revert the changes here to the docstrings, since json strings are still valid. And it's not overly obvious to someone that yaml is a superset of json.

**Review by mattrobenolt — APPROVED** [review]

**Review by mattrobenolt — APPROVED** [review]

**Comment by joshuarli** [comment]
I.... cannot reproduce the following locally on 5.1, 5.2, or 5.3. Travis is running on 5.3 and failing. I can reproduce this locally on 3.11 which makes sense.

```
File "/home/travis/build/getsentry/sentry/tests/sentry/utils/test_types.py", line 76, in test_dict

    assert Dict("{foo:bar}") == {"foo:bar": None}

  File "/home/travis/build/getsentry/sentry/src/sentry/utils/types.py", line 39, in __call__

    raise InvalidTypeError(u"{!r} is not a valid {}".format(value, repr(self)))

InvalidTypeError: '{foo:bar}' is not a valid dictionary
```

**Comment by joshuarli** [comment]
Well well well... turns out it was libyaml after all. I built PyYAML 5.3 with libyaml 0.2.2 from brew, and was able to reproduce the failure. Same thing with 5.1 wheel with libyaml 0.2.2.

Also this suggests travis xenial ships with libyaml already, which is nice so we don't have to do something like https://github.com/getsentry/sentry/pull/16923/commits/898933ec94349b46219f41fda56a4aacf83de298.

**Comment by joshuarli** [comment]
Wait. I squinted, and locally:

```
Traceback (most recent call last):
  File "/Users/joshua.li/dev/sentry/sentry/tests/sentry/utils/test_types.py", line 77, in test_dict
    assert Dict("{foo:}") == {"foo": None}
  File "/Users/joshua.li/dev/sentry/sentry/src/sentry/utils/types.py", line 39, in __call__
    raise InvalidTypeError(u"{!r} is not a valid {}".format(value, repr(self)))
InvalidTypeError: '{foo:}' is not a valid dictionary
```

That doesn't match with CI's `InvalidTypeError: '{foo:bar}' is not a valid dictionary`.

**Review by mattrobenolt — APPROVED** [review]
